src/mlpack/methods/ann/layer/binary_rbm_impl.hpp

```diff
@@ -19,61 +19,63 @@ namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
 template<typename InputDataType, typename OutputDataType>
-VisibleLayer<InputDataType, OutputDataType>::VisibleLayer(
+BinaryLayer<InputDataType, OutputDataType>::BinaryLayer(
   const size_t inSize,
-	const size_t outSize):
+	const size_t outSize,
+  const bool typeVisible):
   inSize(inSize), 
-  outSize(outSize)
+  outSize(outSize),
+  typeVisible(typeVisible)
 {
   weights.set_size(outSize * inSize + inSize, 1);
 }
 
 template<typename InputDataType, typename OutputDataType>
-void VisibleLayer<InputDataType, OutputDataType>::Reset()
+void BinaryLayer<InputDataType, OutputDataType>::Reset()
 {
-  
-  weight = arma::mat(weights.memptr(), outSize, inSize, false, false);
-  ownBias = arma::mat(weights.memptr() + weight.n_elem, inSize, false, false);
-  otherBias = arma::mat(weight.memptr() + weight.n_elem + inSize, outSize, false, false);
-
+  if(typeVisible)
```

Comment:
I think `inSize` should denote the number of input units. I guess if the meaning of `inSize` depends on the type of the level it is confusing.